src/PKSim.Core/Snapshots/Mappers/QualificationStepMapper.cs

```diff
@@ -0,0 +1,91 @@
+
+      public override async Task<SnapshotQualificationStep> MapToSnapshot(IQualificationStep qualificationStep)
+      {
+         var snapshot = await SnapshotFrom(qualificationStep, x => { x.Type = friendlyNameFor(qualificationStep.GetType().Name); });
+         mapQualificationStepPropertiesToSnapsot(snapshot, qualificationStep);
+         return snapshot;
+      }
+
+      private void mapQualificationStepPropertiesToSnapsot(SnapshotQualificationStep snapshot, IQualificationStep qualificationStep)
+      {
+         switch (qualificationStep)
+         {
+            case RunParameterIdentificationQualificationStep runParameterIdentification:
+               snapshot.Subject = runParameterIdentification.ParameterIdentification.Name;
+               break;
+            case RunSimulationQualificationStep runSimulationQualificationStep:
+               snapshot.Subject = runSimulationQualificationStep.Simulation.Name;
+               break;
+            default:
+               throw new ArgumentException(PKSimConstants.Error.CouldNotFindQualificationStep(snapshot.Type));
+         }
+      }
+
+      private string friendlyNameFor(string qualificationStepType) => qualificationStepType.Replace(QUALIFICATION_STEP_SUFFIX, "");
+
+      public override Task<IQualificationStep> MapToModel(SnapshotQualificationStep snapshot, PKSimProject project)
+      {
+         var qualificationStep = createQualificationStepFrom(snapshot.Type);
+         if (qualificationStep == null)
+         {
+            _logger.AddWarning(PKSimConstants.Error.CouldNotFindQualificationStep(snapshot.Type));
+            return Task.FromResult<IQualificationStep>(null);
+         }
+
+         mapQualificationStepPropertiesToModel(qualificationStep, snapshot, project);
+
+         return Task.FromResult(qualificationStep);
+      }
+
+      private void mapQualificationStepPropertiesToModel(IQualificationStep qualificationStep, SnapshotQualificationStep snapshot, PKSimProject project)
+      {
+         switch (qualificationStep)
+         {
+            case RunParameterIdentificationQualificationStep runParameterIdentification:
+               var parameterIdentification = project.AllParameterIdentifications.FindByName(snapshot.Subject);
+               if (parameterIdentification == null)
+                  throw new SnapshotOutdatedException(PKSimConstants.Error.CouldNotFindParameterIdentification(snapshot.Subject));
+
+               runParameterIdentification.ParameterIdentification = parameterIdentification;
+               break;
+            case RunSimulationQualificationStep runSimulationQualificationStep:
+               var simulation = project.All<Model.Simulation>().FindByName(snapshot.Subject);
+               if (simulation == null)
+                  throw new SnapshotOutdatedException(PKSimConstants.Error.CouldNotFindSimulation(snapshot.Subject));
+
+               runSimulationQualificationStep.Simulation = simulation;
+               break;
+         }
```

Comment:
Missing `default:` path

Reply:
don't agree. There might not be anything to map for some types. Also the line 47 already checks that the type exists.

Reply:
The type exists mut might not be handled properly if we introduce new types of qualification steps

Reply:
lol. YEah in that case the developer did not even try it once or wrote tests. Unlikely.
But OK I will throw an exception in that case